src/plugins/advanced_settings/public/header_user_theme_menu.tsx

```diff
@@ -0,0 +1,206 @@
+    (window.localStorage.getItem('useBrowserColorScheme') && window.matchMedia) || false;
+  const [darkMode, setDarkMode] = useUiSetting$<boolean>('theme:darkMode');
+  const [themeVersion, setThemeVersion] = useUiSetting$<string>('theme:version');
+  const [isPopoverOpen, setPopover] = useState(false);
+  // TODO: improve naming?
+  const [theme, setTheme] = useState(themeOptions.find((t) => t.text === themeVersion)?.value);
+  const [screenMode, setScreenMode] = useState(
+    prefersAutomatic
+      ? screenModeOptions[2].value
+      : darkMode
+      ? screenModeOptions[1].value
+      : screenModeOptions[0].value
+  );
+  const allSettings = uiSettings.getAll();
+  const defaultTheme = allSettings['theme:version'].value;
+  const defaultScreenMode = allSettings['theme:darkMode'].value;
+
+  const onButtonClick = () => {
+    setPopover(!isPopoverOpen);
+  };
+
+  const onThemeChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    setTheme(e.target.value);
+  };
+
+  const onScreenModeChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    setScreenMode(e.target.value);
+  };
+
+  const onAppearanceSubmit = async (e: SyntheticEvent) => {
+    const actions = [setThemeVersion(themeOptions.find((t) => theme === t.value)?.text ?? '')];
+
+    if (screenMode === 'automatic') {
+      const browserMode = window.matchMedia('(prefers-color-scheme: dark)').matches;
+      window.localStorage.setItem('useBrowserColorScheme', 'true');
+
+      if (browserMode !== darkMode) {
+        actions.push(setDarkMode(browserMode));
+      }
+    } else if ((screenMode === 'dark') !== darkMode) {
+      actions.push(setDarkMode(screenMode === 'dark'));
+      window.localStorage.removeItem('useBrowserColorScheme');
+    } else {
+      window.localStorage.removeItem('useBrowserColorScheme');
+    }
+    // TODO: only set changed
+    await await Promise.all([actions]);
+    window.location.reload();
+  };
+
+  const closePopover = () => {
+    setPopover(false);
+  };
+
+  const button = (
+    <EuiHeaderSectionItemButton
+      aria-expanded="false"
+      aria-haspopup="true"
+      aria-label={i18n.translate('advancedSettings.headerGlobalNav.themeMenuButtonAriaLabel', {
+        defaultMessage: 'Appearance menu',
+      })}
+      onClick={onButtonClick}
+    >
+      <EuiIcon
+        type="color"
+        size="m"
+        title={i18n.translate('advancedSettings.headerGlobalNav.themeMenuButtonTitle', {
+          defaultMessage: 'Appearance',
+        })}
+      />
+    </EuiHeaderSectionItemButton>
+  );
+
+  // TODO: make i18n, check all translation ids
+  // TODO: fix focus behavior
+  const appearanceContent = (
+    <div style={{ maxWidth: 300 }}>
+      <EuiCallOut color="warning">
+        These settings apply to only this user account. To change settings for the entire
+        application, {/* TODO: use navigateToApp instead? */}
+        <EuiLink
+          href={basePath.prepend('/app/management/opensearch-dashboards/settings#appearance')}
```

Comment:
Or use `getUrlForApp`

Reply:
Right, thanks for the reminder! I think this message may be removed altogether, now that the advanced settings page actually does the exact same thing (update for that browser, not for the whole cluster).

Reply:
added to check https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5652/files#r1460030358

cc: @kgcreative